src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -9,91 +10,281 @@ import {
+          initialize: {
+            titleTKey: "Provisioning resource",
+            messageTKey: "Dedicated Gateway resource will be provisioned.",
+          },
+          success: {
+            titleTKey: "Resource provisioned",
+            messageTKey: `Dedicated Gateway resource provisioned. Please go to <a href='${generateBladeLink(
+              BladeType.SqlKeys
+            )}'>keys blade</a> to use the keys.`,
+          },
+          failure: {
+            titleTKey: "Provisioning resource failed.",
+            messageTKey: "Dedicated Gateway resource provisioning failed.",
+          },
+        },
+      };
+    }
   };
 
   public initialize = async (): Promise<Map<string, SmartUiInput>> => {
-    // TODO: get initialization data from initializeDedicatedGatewayProvisioning() RP call.
-    throw new Error("onSave not implemented");
+    // Based on the RP call enableDedicatedGateway will be true if it has not yet been enabled and false if it has.
+    const defaults = new Map<string, SmartUiInput>();
+    defaults.set("enableDedicatedGateway", { value: false });
+    defaults.set("sku", { value: "Cosmos.D4s", hidden: true });
+    defaults.set("instances", { value: await getInstancesMin(), hidden: true });
+    defaults.set("skuDetails", undefined);
+    defaults.set("costPerHour", undefined);
+
+    const response = await getCurrentProvisioningState();
+    if (response.status && response.status !== "Deleting") {
+      defaults.set("enableDedicatedGateway", { value: true });
+      defaults.set("sku", { value: response.sku, disabled: true });
+      defaults.set("instances", { value: response.instances, disabled: true });
+      defaults.set("costPerHour", { value: costPerHourValue });
+      defaults.set("skuDetails", {
+        value: { textTKey: getSKUDetails(`${defaults.get("sku").value}`), type: DescriptionType.Text } as Description,
+        hidden: false,
+      });
+    }
+
+    defaults.set("warningBanner", undefined);
+    return defaults;
   };
 
+  @Values({
+    isDynamicDescription: true,
+  })
+  warningBanner: string;
+
   @Values({
     description: {
-      textTKey: "Provisioning dedicated gateways for SqlX accounts.",
+      textTKey: "DedicatedGatewayDescription",
       type: DescriptionType.Text,
       link: {
         href: "https://docs.microsoft.com/en-us/azure/cosmos-db/introduction",
-        textTKey: "Learn more about dedicated gateway.",
+        textTKey: "LearnAboutDedicatedGateway",
       },
     },
   })
   description: string;
 
   @OnChange(onEnableDedicatedGatewayChange)
   @Values({
-    labelTKey: "Dedicated Gateway",
+    labelTKey: "DedicatedGateway",
     trueLabelTKey: "Enable",
     falseLabelTKey: "Disable",
   })
   enableDedicatedGateway: boolean;
 
+  @OnChange(onSKUChange)
   @Values({
     labelTKey: "SKUs",
     choices: getSkus,
-    placeholderTKey: "Select SKUs",
+    placeholderTKey: "SKUsPlaceHolder",
   })
   sku: ChoiceItem;
 
   @Values({
```

Comment:
We need to add another dynamic description that has a reference to the keys blade